### PR TITLE
fixed bug in prepare function and added trend and constant columns

### PR DESCRIPTION
**Pull request by ll01**
I think this is what is needed, however now some of the matrix are failing to invert in other tests. I know the python version uses sudo invert, but I don't know how that will affect performance.

### CODE REVIEW

**Comment by ll01** [comment]
#3

**Comment by ssoudan** [comment]
Thanks and apologies for misguiding you about the column of `1`: 

After a bit of digging, the column of `1` was actually already added in the `ols` code (https://github.com/ssoudan/unit-root/blob/190750e7ecff43779da959170ccf97fad2ed5b96/src/regression.rs#L28). The code without the change is already doing the 'constant no trends' - similar to statsmodels's default. 

Took me some time to figure out why I used `t_stats[1]`, in https://github.com/ssoudan/unit-root/blob/190750e7ecff43779da959170ccf97fad2ed5b96/src/tools/adf.rs#L36 .

The test statistics is the approximate coefficient of the y[t-1] divided by its SE (https://en.wikipedia.org/wiki/Augmented_Dickey–Fuller_test#Testing_procedure).

The column of `1` was added in the first position, so the \hat{y}[t-1] is in the second coefficient. 

Bottom line, if we remove the constant column from the ols() function it is not longer an ols. I think the named should be changed to `regress` or something like that. The method is public but only exposed when the feature 'unstable' is used - and it it intended to be used only by the benchmarks. Then depending on where the y[t-1] is, the right test statistics need to be pulled.

File: src/tools/dickeyfuller.rs [hunk 1]
```diff
@@ -76,12 +77,12 @@ use crate::Error;
 pub fn constant_no_trend_test<F: Float + Scalar + RealField>(
     series: &DVector<F>,
 ) -> Result<Report<F>, Error> {
-    let (delta_y, y_t_1, size) = prepare(series, 0)?;
+    let (delta_y, y_t_1, size) = prepare(series, 0, Regression::Constant)?;
 
     let (_betas, t_stats) = ols(&delta_y, &y_t_1)?;
-
+    println!("t_stats: {}", t_stats);
```
Comment (ssoudan) [inline src/tools/dickeyfuller.rs:83]: -println

**Review by ssoudan — CHANGES_REQUESTED** [review]
Looking good! Once the 'println' is removed, I'm thinking of merging this PR and you can create another one to expose the new tests that the changes you introduced allow. Works for you?

**Review by ssoudan — APPROVED** [review]

**Comment by ssoudan** [comment]
Thanks!
